### PR TITLE
[186] Add transaction sync status to orders

### PR DESCRIPTION
**Pull request by Noah-Silvera**
What is the goal of this PR?
---

We should display the status of the most recent transaction on the status of an ordwr

How do you manually test these changes? (if applicable)
---

1. Report an order to taxjar
    * [x] Ensure the date and time of the latest sync is shown on the order page
    * [x] Ensure the status of the order sync is shown on the order page

Merge Checklist
---

- [x] Run the manual tests
- [x] Update the changelog

Screenshots
---
Screen Shot 2022-10-03 at 11 21 25PDT
Screen Shot 2022-10-03 at 11 21 11PDT
Screen Shot 2022-10-03 at 11 20 49PDT
Screen Shot 2022-10-03 at 11 20 40PDT
Screen Shot 2022-10-03 at 11 19 45PDT

### CODE REVIEW

File: spec/features/spree/admin/reporting_to_taxjar_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    create :store, default: true
+    create :taxjar_configuration, :reporting_enabled
+  end
+
+  let!(:order) { create :order_ready_to_ship }
+
+  scenario "shipping a complete and paid order", vcr: { allow_unused_http_interactions: false } do
+    visit spree.edit_admin_order_path(order)
+    perform_enqueued_jobs do
+      expect(page).not_to have_content("Loading")
```
Comment (benjaminwil) [inline spec/features/spree/admin/reporting_to_taxjar_spec.rb:18]: (Maybe ignore this comment. I am nitpicking.)

As a Rails developer, I understand what is implied by this expectation and why we might want to expect it. But I think the way it's written doesn't really map to something a user would do. (I.e. users don't wait for a page to stop saying loading. They wait for a page to finish loading.)

Maybe this is a bad suggestion, but I might write this this way to make the intent more clear:

```
  # ...
  visit spree.edit_admin_order_path(order)
  perform_enqueued_jobs do
    wait_for_order_information_to_load
    # ...
  end
end
    
def wait_for_order_information_to_load
  expect(page).not_to have_content "Loading"
end
```

File: app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface [hunk 2]
```diff
@@ -1,8 +1,30 @@
 
 <!-- insert_bottom "[id='order_tab_summary'] > .additional-info" -->
 
+<% content_for :head do %>
+  <style>
+    .pill-success {
+      background: #c8d8e8;
+    }
+  </style>
```
Comment (benjaminwil) [inline app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface:9]: I am surprised we need this. Bootstrap provides a `.pill-success` class:

https://getbootstrap.com/docs/4.0/components/badge/#pill-badges

**Review by benjaminwil — APPROVED** [review]

File: app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface (hunk 2 above)
Comment (Noah-Silvera) [reply app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface:9]: Huh!

This is where the pill classes come from in solidus! I don't think they use the bootstrap classes
https://github.com/solidusio/solidus/blob/master/backend/app/assets/stylesheets/spree/backend/components/_pills.scss

File: spec/features/spree/admin/reporting_to_taxjar_spec.rb (hunk 1 above)
Comment (Noah-Silvera) [reply spec/features/spree/admin/reporting_to_taxjar_spec.rb]: It's a good nit! Thanks for the reminder as I haven't been doing this in my tests lately.

It does make the intent more clear

Comment (Noah-Silvera) [reply spec/features/spree/admin/reporting_to_taxjar_spec.rb]: Updated!

**Review by Noah-Silvera — COMMENTED** [review]
Addressed @benjaminwil's comment

File: app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface (hunk 2 above)
Comment (benjaminwil) [reply app/overrides/spree/admin/shared/_order_summary/add_taxjar_reported_at.html.erb.deface:9]: Wild! All this time I thought we used Bootstrap in the admin for that kind of stuff.

**Review by adammathys — APPROVED** [review]

**Review by nvandoorn — APPROVED** [review]

**Comment by Noah-Silvera** [comment]
Just a rebase on master in prep for merge!

**Pull request merged**